pyro/distributions/householder.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+import warnings
+
+import torch
+import torch.nn as nn
+from torch.distributions import constraints
+
+from pyro.distributions.torch_transform import TransformModule
+from pyro.distributions.util import copy_docs_from
+
+
+@copy_docs_from(TransformModule)
+class HouseholderFlow(TransformModule):
+    """
+    A single transformation of Householder flow,
+
+        :math:`\\mathbf{y} = (I - 2*\\frac{\\mathbf{u}\\mathbf{u}^T}{||\\mathbf{u}||^2})\\mathbf{x}`
+
+    where :math:`\\mathbf{x}` are the inputs, :math:`\\mathbf{y}` are the outputs, and the learnable parameters
+    are :math:`\\mathbf{u}\\in\\mathbb{R}^D` for input dimension :math:`D`.
+
+    The transformation represents the reflection of :math:`\\mathbf{x}` through the plane passing through the
+    origin with normal :math:`\\mathbf{u}`. Together with `TransformedDistribution` this provides a way to
+    create richer variational approximations.
+
+    :math:`D` applications of this transformation are able to transform standard i.i.d. standard Gaussian noise
+    into a Gaussian variable with an arbitrary covariance matrix. With :math:`K<D` transformations, one is able
+    to approximate a full-rank covariance Gaussian distribution.
+
+    Example usage:
+
+    >>> base_dist = dist.Normal(torch.zeros(10), torch.ones(10))
+    >>> flows = [HouseholderFlow(10) for _ in range(3)]
+    >>> [pyro.module("my_flow", p) for f in flows] # doctest: +SKIP
```

Comment:
update docstring to create a single flow with multiple reflections?

Reply:
Done!